### PR TITLE
runSimulationBatches returns a named list

### LINKED ISSUE #1031: `runSimulationBatches` does not return a named list
The documentation states

> #' @return Nested list of `SimulationResults` objects. The first level of the list are the IDs of the simulations of SimulationBatches, containing a list of `SimulationResults` for each set of parameter/initial values. If a simulation with a parameter/initial values set fails, the result for this run is `NULL`

However, this should be the IDs of the batches and not the simulations (as multiple batches can be created from the same simulation). Variable naming is confusing and the documentation.

### PR DESCRIPTION
**Pull request by PavelBal**
Fixes #1031

### CODE REVIEW

File: R/utilities-simulation.R [hunk 1]
```diff
@@ -374,35 +377,38 @@ runSimulationBatches <- function(simulationBatches, simulationRunOptions = NULL,
...
 
   simulationBatches <- c(simulationBatches)
-  # Result Id <-> simulation batch pointer id map to get the correct simulation for the results.
-  # Using the Id of the pointer instead of the Id of the simulation as multiple
+  # Result Id <-> simulation batch id map to get the correct simulation for the results.
+  # Using the Id of the batch instead of the Id of the simulation as multiple
   # SimulationBatches can be created with the same simulation
   # Each SimulationBatchRunValues has its own id, which will be the id of the result
-  resultsIdSimulationIdMap <- list()
+  resultsIdSimulationBatchIdMap <- list()
```
Comment (PavelBal) [inline R/utilities-simulation.R:384]: Renamed variables to reflect the actual code.

File: R/simulation-batch.R [hunk 2]
```diff
@@ -110,6 +111,10 @@ SimulationBatch <- R6::R6Class(
       } else {
         private$throwPropertyIsReadonly("runValuesIds")
       }
+    },
+    #' @field id The id of the .NET wrapped object. (read-only)
```
Comment (IndrajeetPatil) [inline R/simulation-batch.R:115]:
```suggestion
    #' @field id A unique identifier for the `.NET` wrapped object. (read-only)
```

**Review by IndrajeetPatil — CHANGES_REQUESTED** [review]
You can delete the new image in `tests/testthat/_snaps/plot-population-time-profile`.

What's up with the two new image files (`figure.png` and `figure_.png`) that have been introduced in `tests/testthat` folder?

File: R/simulation-batch.R (hunk 2 above)
Comment (msevestre) [reply R/simulation-batch.R:115]: it sounds like it's not the one from .net where in fact, it IS

File: R/utilities-quantity.R [hunk 3]
```diff
@@ -211,20 +215,24 @@ getQuantityValuesByPath <- function(quantityPaths, simulation, units = NULL, sto
...
-      #in necessary
-      if (!is.null(units[[i]])){
+      # If the unit is NULL, the value is assumend to be in base unit and no conversion
+      # in necessary
+      if (!is.null(units[[i]])) {
         mw <- simulation$molWeightFor(path)
-        value <- toUnit(quantityOrDimension = dimension, values = value, targetUnit = units[[i]],
-                        molWeight = mw)
+        value <- toUnit(
+          quantityOrDimension = dimension, values = value, targetUnit = units[[i]],
```
Comment (msevestre) [inline R/utilities-quantity.R:233]: same here

File: R/utilities-quantity.R [hunk 4]
```diff
@@ -159,12 +159,14 @@ setQuantityValuesByPath <- function(quantityPaths, values, simulation, units = N
...
-      #in necessary
-      if (!is.null(units[[i]])){
+      # If the unit is NULL, the value is assumend to be in base unit and no conversion
+      # in necessary
+      if (!is.null(units[[i]])) {
         mw <- simulation$molWeightFor(path)
-        value <- toBaseUnit(quantityOrDimension = dimension, values = value, unit = units[[i]],
-                            molWeight = mw)
+        value <- toBaseUnit(
+          quantityOrDimension = dimension, values = value, unit = units[[i]],
```
[the remark this answers is not in the document]
Comment (PavelBal) [reply R/utilities-quantity.R:167]: Styler, I didn't change anything manually.

File: R/utilities-simulation.R [hunk 5]
```diff
@@ -374,35 +377,38 @@ runSimulationBatches <- function(simulationBatches, simulationRunOptions = NULL,
...
 
   # Run the batch with the ConcurrentSimulationRunner
   results <- rClr::clrCall(simulationRunner, "RunConcurrently")
-  simulationResults <- .getConcurrentSimulationRunnerResults(results = results, resultsIdSimulationIdMap = resultsIdSimulationIdMap, simulationIdSimulationMap = simulationIdSimulationMap, silentMode = silentMode)
-
-  # output: list of lists of SimulationResults, one list per SimulationBatch
-  output <- lapply(names(simulationIdSimulationMap), function(simId) {
-    simulationResults[which(resultsIdSimulationIdMap == simId)]
-  })
+  simulationResults <- .getConcurrentSimulationRunnerResults(results = results, resultsIdSimulationIdMap = resultsIdSimulationBatchIdMap, simulationIdSimulationMap = simulationBatchIdSimulationMap, silentMode = silentMode)
```
Comment (msevestre) [inline R/utilities-simulation.R:403]: one parameter by line here as well. I need to scroll to see the code

File: R/simulation-batch.R (hunk 2 above)
Comment (PavelBal) [reply R/simulation-batch.R:115]: I would leave it as "ID" as this is how the field in .NET is called..

File: R/utilities-simulation.R [hunk 6]
```diff
@@ -374,35 +377,38 @@ runSimulationBatches <- function(simulationBatches, simulationRunOptions = NULL,
...
+
+  # Returned is a named list of results with names being the IDs of the batches
+  output <- vector("list", length(simulationBatches))
+  names(output) <- names(simulationBatchIdSimulationMap)
+  # Iterate through simulation batch ids
+  for (simBatchId in names(simulationBatchIdSimulationMap)) {
```
Comment (msevestre) [inline R/utilities-simulation.R:409]: Can't you ued the output from before and set the name after the fact? creating the vector and a for seems so not R like :)

**Review by msevestre — CHANGES_REQUESTED** [review]
minor changes

File: R/utilities-quantity.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (PavelBal) [reply R/utilities-quantity.R]: Re-styled

File: R/utilities-simulation.R (hunk 6 above)
Comment (PavelBal) [reply R/utilities-simulation.R]: Reverted and just added assigning names :)

**Review by IndrajeetPatil — APPROVED** [review]

**Pull request merged**